File: BOJ_solved.ac/CLASS_3/Silver_1541.py
# 잃어버린 괄호
# 최솟값을 만들기 위해서는 - 기준으로 괄호를 치면 된다.
# 예를 들어 55 - 50 + 40 - 30 + 20 위와 같이 입력을 받았을때 - 기준으로 55 - (50 + 40) - (30 + 20) 이렇게 괄호를 쳤을때 최솟값이 된다.

import sys

input = sys.stdin.readline
exp = input()

def solution(exp):
    memory = 0
    forsum = []
    result = []
    for i in range(len(exp)):
        if exp[i] == '+' or exp[i] == '-' or i == len(exp)-1:
            forsum.append(int(exp[memory:i]))
            memory = i+1
            if exp[i] == '-':
                result.append(sum(forsum))
                forsum = []
    if len(forsum) != 0:
        result.append(sum(forsum))
    
    answer = 0
    for i in range(len(result)):
        if i == 0:
            answer = result[i]
            continue
        answer -= result[i]

    return answer

print(solution(exp))

File: BOJ_solved.ac/CLASS_3/Silver_1541.java
// 잃어버린 괄호

import java.util.*;

public class Main {
    public static int sumall(ArrayList<Integer> forsum) {
        int sum = 0;
        for(int i : forsum) {
            sum += i;
        }
        return sum;
    }

    public static ArrayList solution(String exp) {
        int memory = 0;
        ArrayList<Integer> forsum = new ArrayList<Integer>();
        ArrayList<Integer> result = new ArrayList<Integer>();

        for(int i=0; i<exp.length(); i++) {
            if(exp.charAt(i) == '+' || exp.charAt(i) == '-') {
                forsum.add(Integer.parseInt(exp.substring(memory, i)));
                memory = i+1;
                if(exp.charAt(i) == '-') {
                    int sum = sumall(forsum);
                    result.add(sum);
                    forsum.clear();
                }
            }
            if(i == exp.length()-1) {
                forsum.add(Integer.parseInt(exp.substring(memory, i+1)));
            }
        }
        int sum = sumall(forsum);
        result.add(sum);
        return result;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String exp = sc.nextLine();

        ArrayList<Integer> result = solution(exp);

        int answer = result.get(0);
        for(int i = 1; i < result.size(); i++){
            answer -= result.get(i);
        }
        System.out.println(answer);
    }
}
